src/main/scala/calc/Tree.scala

```diff
@@ -2,25 +2,29 @@ package calc
 
 import org.scalajs.core.ir.Position
 
+
 sealed abstract class Tree {
   def pos: Position
+  def tp: TreeType
 }
 
-final case class Literal(value: Double)(implicit val pos: Position) extends Tree
+final case class Literal(value: Double, tp: TreeType=NoType)
+                        (implicit val pos: Position) extends Tree
```

Comment:
Hum ... You're not supposed to modify the untyped AST.
It would be better to replicate the trees in a separate hierarchy for typed ASTs, like `TypedTree`, `TypedLiteral`, etc.